grc/core/utils/extract_docs.py

```diff
@@ -185,14 +185,16 @@ def _handle_worker(self):
     def _send(self, cmd, args):
         """ Send a command to worker's stdin """
         fd = self._worker.stdin
-        json.dump((self.AUTH_CODE, cmd, args), fd)
-        fd.write('\n'.encode())
+        query = json.dumps((self.AUTH_CODE, cmd, args))
+        fd.write(query.encode('utf-8'))
+        fd.write(b'\n')
+        fd.flush()
 
     def _receive(self):
         """ Receive response from worker's stdout """
         for line in iter(self._worker.stdout.readline, ''):
             try:
-                key, cmd, args = json.loads(line, encoding='utf-8')
+                key, cmd, args = json.loads(line.decode('utf-8'), encoding='utf-8')
```

Comment:
We don't need the `encoding='utf-8'` part anymore. Also, I just learned that its use is deprecated anyway https://docs.python.org/3/library/json.html#json.loads =)